File: front_end/src/service/KhachHangService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/khach-hang';

export const getAllKhachHang = (pageable, keyword) => {
    return axios.get(API_URL, {
        params: {
            page: pageable.page,
            size: pageable.size,
            keyword: keyword
        }
    });
};

export const createKhachHang = (khachHangRequest) => axios.post(API_URL, khachHangRequest);

export const getOneKhachHang = (khachHangId) => axios.get(`${API_URL}/${khachHangId}`);

export const updateKhachHang = (khachHangId, khachHangRequest) => 
    axios.put(`${API_URL}/${khachHangId}`, khachHangRequest);

export const deleteKhachHang = (khachHangId) => axios.delete(`${API_URL}/${khachHangId}`);

File: src/main/java/com/example/shoe/service/KhachHangService.java
package com.example.shoe.service;

import com.example.shoe.dto.request.KhachHangRequest;
import com.example.shoe.dto.response.KhachHangResponse;
import com.example.shoe.entity.KhachHang;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface KhachHangService {
    Page<KhachHangResponse> getAllKhachHang(Pageable pageable, String keyword);
    KhachHangResponse createKhachHang(KhachHangRequest request);
    KhachHangResponse getOneKhachHang(Integer khachHangId);
    KhachHangResponse updateKhachHang(Integer khachHangId, KhachHangRequest request);
    Boolean deleteKhachHang(Integer khachHangId);
    KhachHang getOrCreateKhachLe();
}
